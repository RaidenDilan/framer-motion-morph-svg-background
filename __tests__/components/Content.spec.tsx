import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Content from '../../src/components/Content';

describe(__filename, () => {
  afterEach(cleanup);

  it('should render without crashing', () => {
    const { asFragment } = render(<Content />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with Content as className', () => {
    const { getByTestId } = render(<Content />);
    const content = getByTestId('Content');
    expect(content).toHaveClass('content content--fixed');
  });

  it('should render child elements', () => {
    const { getByTestId } = render(<Content />);
    const codropsHeaderTitle = getByTestId('codrops-header__title');
    expect(codropsHeaderTitle).toHaveClass('codrops-header__title');

    const demoOne = getByTestId('demo-1');
    expect(demoOne).toHaveClass('demo');

    const demoTwo = getByTestId('demo-2');
    expect(demoTwo).toHaveClass('demo');

    const demoThree = getByTestId('demo-3');
    expect(demoThree).toHaveClass('demo');

    const paterImg = getByTestId('pater__img');
    expect(paterImg).toHaveClass('pater__img');
    expect(paterImg).toHaveAttribute('data-testid', 'pater__img');
    expect(paterImg).toHaveAttribute('alt', 'HelloSign API');

    const paterTitle = getByTestId('pater__title');
    expect(paterTitle).toHaveClass('pater__title');
    expect(paterTitle).toHaveTextContent('2x Faster Integration Time');

    const paterDesc = getByTestId('pater__desc');
    expect(paterDesc).toHaveClass('pater__desc');
    expect(paterDesc).toHaveTextContent(
      'Embed contracts and agreements into your site or app with a few lines of code.',
    );

    const decoTitle = getByTestId('deco__title');
    expect(decoTitle).toHaveClass('deco deco--title');
    expect(decoTitle).toHaveTextContent('2017 Westland Exhibition Center');
  });
});
