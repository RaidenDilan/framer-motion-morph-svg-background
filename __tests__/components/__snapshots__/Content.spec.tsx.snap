// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/Raiden/development/ANIMATIONS/framer-motion-morph-svg-background/__tests__/components/Content.spec.tsx should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="content content--fixed"
    data-testid="Content"
  >
    <header
      class="codrops-header"
    >
      <div
        class="codrops-links"
      >
        <svg
          class="decoshape"
          height="100%"
          preserveAspectRatio="none"
          viewBox="0 0 200 100"
          width="100%"
        >
          <path
            d="M 10.45,74.41 C 39.4,110.8 72.59,67.27 95.98,68.68 122.3,70.35 131.5,101.4 154.7,99.69 177.9,98.03 200.5,79.26 198.1,47.17 195.9,15.27 174.6,-0.3279 151.8,0.9941 128.6,2.581 126,16.86 107,22.76 88.26,28.67 88.87,12.36 60.37,1.787 31.79,-8.877 -18.61,37.92 10.45,74.41 Z"
          />
        </svg>
        <a
          class="codrops-icon codrops-icon--prev"
          href="https://tympanus.net/Development/FolderPreviewIdeas/"
          title="Previous Demo"
        >
          <svg
            class="icon icon--arrow"
          >
            <use
              xlink:href="#icon-arrow"
            />
          </svg>
        </a>
        <a
          class="codrops-icon codrops-icon--drop"
          href="https://tympanus.net/codrops/?p=31190"
          title="Back to the article"
        >
          <svg
            class="icon icon--drop"
          >
            <use
              xlink:href="#icon-drop"
            />
          </svg>
        </a>
      </div>
      <h1
        class="codrops-header__title"
        data-testid="codrops-header__title"
      >
        Morphing Background Shapes
      </h1>
    </header>
    <nav
      class="demos"
    >
      <a
        class="demo"
        data-testid="demo-1"
        href="index.html"
      >
        <svg
          class="decoshape"
          preserveAspectRatio="none"
        >
          <use
            xlink:href="#shape-demo"
          />
        </svg>
        Demo 1
      </a>
      <a
        class="demo demo--current"
        data-testid="demo-2"
        href="index2.html"
      >
        <svg
          class="decoshape"
        >
          <use
            xlink:href="#shape-demo"
          />
        </svg>
        Demo 2
      </a>
      <a
        class="demo"
        data-testid="demo-3"
        href="index3.html"
      >
        <svg
          class="decoshape"
        >
          <use
            xlink:href="#shape-demo"
          />
        </svg>
        Demo 3
      </a>
    </nav>
    <a
      class="pater"
      href="http://synd.co/2pHxJny"
    >
      <svg
        class="pater__deco"
        height="240"
        viewBox="0 0 1000 800"
        width="300"
      >
        <path
          d="M27.4,171.8C73,42.9,128.6,1,128.6,1s0,0,0,0c58.5,0,368.3,0.3,873.2,0.8c38.5,211,42.1,373.5,38.9,476.7c-2.5,80.3-10.6,174.9-76.7,247.8c-15.1,16.6-37.4,41.2-72.8,53.9c-92.4,33.1-173-50.8-283.9-99.4c-224.3-98.4-334.9,51.4-472.2-45.6C-6.3,535.2-14.5,290.6,27.4,171.8z"
        />
      </svg>
      <img
        alt="HelloSign API"
        class="pater__img"
        data-testid="pater__img"
        src="[object Object]"
      />
      <h4
        class="pater__title"
        data-testid="pater__title"
      >
        2x Faster Integration Time
      </h4>
      <p
        class="pater__desc"
        data-testid="pater__desc"
      >
        Embed contracts and agreements into your site or app with a few lines of code.
      </p>
    </a>
    <div
      class="deco deco--title"
      data-testid="deco__title"
    >
      2017 Westland Exhibition Center
    </div>
  </div>
</DocumentFragment>
`;
